activation/activation.go

```diff
@@ -261,19 +261,28 @@ func (b *Builder) run(ctx context.Context) {
 	b.loop(ctx)
 }
 
-func (b *Builder) generateProof(ctx context.Context) error {
-	// don't generate the commitment every time smeshing is starting, but once only.
-	if _, err := b.cdb.GetLastAtx(b.nodeID); err != nil {
-		// Once initialized, run the execution phase with zero-challenge,
-		// to create the initial proof (the commitment).
-		startTime := time.Now()
-		b.initialPost, b.initialPostMeta, err = b.postSetupProvider.GenerateProof(ctx, shared.ZeroChallenge)
-		if err != nil {
-			return fmt.Errorf("post execution: %w", err)
-		}
-		metrics.PostDuration.Set(float64(time.Since(startTime).Nanoseconds()))
+func (b *Builder) generateInitialPost(ctx context.Context) error {
+	// Generate the initial POST if we don't have an ATX...
+	if _, err := b.cdb.GetLastAtx(b.nodeID); err == nil {
+		return nil
+	}
+	// ...and if we don't have an initial POST persisted already.
+	if _, err := loadPost(b.nipostBuilder.DataDir()); err == nil {
+		return nil
 	}
 
+	// Create the initial post and save it.
+	startTime := time.Now()
+	var err error
```

Comment:
it makes sense to group variables in var ( ... ) block

Reply:
What's the benefit? I prefer to define variables next to the place they are used.

Reply:
`startTime` is initialized while `err` is not. Using `var (...)` adds 2 extra lines w/o any real benefit.
```
startTime := time.Now()
var err error
```
vs.
```go
var (
	startTime = time.Now()
	err       error
)
```

Reply:
does `err` need to be declared explicitly? The next line could just be:

```go
b.initialPost, b.initialPostMeta, err = b.postSetupProvider.GenerateProof(ctx, shared.ZeroChallenge)
```

Reply:
:+1: `:=` doesn't work with fields.